app/models/nic/base.rb

```diff
@@ -146,14 +152,33 @@ def host_managed?
 
     def require_ip_validation?
       # if it's not managed there's nowhere to specify an IP anyway
-      return false if !self.host.managed? || !self.managed? || !self.provision?
+      self.host.managed? && self.managed? && self.provision?
+    end
+
+    def require_ip4_validation?
+      return false unless require_ip_validation?
       # if the CR will provide an IP, then don't validate yet
       return false if host.compute_provides?(:ip)
-      ip_for_dns     = (subnet.present? && subnet.dns_id.present?) || (domain.present? && domain.dns_id.present?)
-      ip_for_dhcp    = subnet.present? && subnet.dhcp_id.present?
-      ip_for_token   = Setting[:token_duration] == 0 && (host.pxe_build? || (host.image_build? && host.image.try(:user_data?)))
+
+      ip_for_dns   = (subnet.present? && subnet.dns_id.present?) || (domain.present? && domain.dns_id.present? && !self.ip6.present?)
+      ip_for_dhcp  = subnet.present? && subnet.dhcp_id.present?
+      ip_for_token = Setting[:token_duration] == 0 && (host.pxe_build? || (host.image_build? && host.image.try(:user_data?))) && !self.ip6.present?
+
       # Any of these conditions will require an IP, so chain with OR
-      ip_for_dns or ip_for_dhcp or ip_for_token
+      ip_for_dns || ip_for_dhcp || ip_for_token
+    end
+
+    def require_ip6_validation?
```

Comment:
This is exactly the same method as `require_ip4_validation?`, can you extract these to a common validator?